File: test/test/ViewController.m
//
//  ViewController.m
//  test
//
//  Created by czljcb on 16/3/13.
//  Copyright © 2016年 czljcb. All rights reserved.
//

#import "ViewController.h"

@interface ViewController ()

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view, typically from a nib.
    
    UILabel *lable = [[UILabel alloc] initWithFrame:CGRectMake(100, 100, 100, 44)];
    lable.text = @"eeeee";
    lable.font = [UIFont systemFontOfSize:45];
    [self.view addSubview:lable];
    
    
}



- (void)injected{
    NSLog(@"I've been injected: %@", self);
    
    
}

- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}

@end

File: 自定义print/自定义print/ViewController.swift
//
//  ViewController.swift
//  自定义Log
//
//  Created by czljcb on 16/2/26.
//  Copyright © 2016年 czljcb. All rights reserved.
//

import UIKit

class ViewController: UIViewController {
    
    override func viewDidLoad() {
        super.viewDidLoad()
        // Do any additional setup after loading the view, typically from a nib.
        
        //        print("\(__FUNCTION__)")
        //
        //        print("\((__FILE__ as NSString).lastPathComponent)")
        //
        //        print("\(__LINE__)")
        
        CZLog("4455")
        test(2,  3)
    }
    
    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }
    
    
}



func test (a : Int , _ b : Int)
{
    
}

func CZLog<T>(message : T,file : String = __FILE__ ,funcName : String = __FUNCTION__, line : Int = __LINE__)
{
    #if DEBUG
        print(" [\((file as NSString).lastPathComponent) -> \(funcName) -> \(line)] \(message)")
    #endif
}
